iphone/Classes/KrollBridge.m

```diff
@@ -1163,18 +1183,25 @@ - (id)require:(KrollContext *)kroll path:(NSString *)path
 
 		// 2. If X begins with './' or '/' or '../'
 		if ([path hasPrefix:@"./"] || [path hasPrefix:@"../"]) {
+
 			// Need base path to work from for relative modules...
 			NSString *workingPath = [oldURL relativePath];
 			NSString *relativePath = (workingPath == nil) ? path : [workingPath stringByAppendingPathComponent:path];
-			module = [self loadAsFileOrDirectory:[relativePath stringByStandardizingPath] withContext:context];
-			if (module) {
+
+			// Load cached require-module if exists
+			module = [self cachedRequireModuleFromPath:relativePath];
+
+			if (module != nil) {
 				return module;
 			}
+
 			// Treat '/' special as absolute, drop the leading '/'
-		}
-		else if ([path hasPrefix:@"/"]) {
-			module = [self loadAsFileOrDirectory:[[path substringFromIndex:1] stringByStandardizingPath] withContext:context];
-			if (module) {
+		} else if ([path hasPrefix:@"/"]) {
+
+			// Load cached require-module if exists
+			module = [self cachedRequireModuleFromPath:path];
```

Comment:
Addressed comments, I didn't respect the different path-structures. One thing I am still wondering: Should `./file` be cached as `file` instead of `./file`? In the following test-case, the first and third require-call will write a new cache (one for `file`, one for `./file`):
```js
var test1 = require('/file');
var test2 = require('/file');
var test3 = require('./file');
var test4 = require('./file');

var win = Ti.UI.createWindow({
    backgroundColor: '#fff'
});

var btn = Ti.UI.createButton({
    title: 'Log Exports'
});

btn.addEventListener('click', function() {
	console.log('#1 ' + test1.hello);
	console.log('#2 ' + test2.hello);
	console.log('#3 ' + test3.hello);
	console.log('#4 ' + test4.hello);
});

win.add(btn);
win.open();
```